File: kornell-api/src/main/scala/kornell/entity/Enrollment.scala
package kornell.entity

import javax.persistence.Column
import javax.persistence.Id
import scala.reflect.BeanProperty
import java.util.Date
import javax.persistence.OneToOne
import javax.persistence.Entity
import java.io.Serializable

@Entity
class Enrollment extends Serializable{
  @BeanProperty
  @Id
  var uuid:String = _
  
  
  @BeanProperty
  var enrolledOn:Date = _
  
  @BeanProperty
  @OneToOne
  var course:Course = _
  
  @BeanProperty
  @OneToOne
  var person:Person = _
  
  override def toString = "Enrollment["+course+","+person+","+enrolledOn+"]"
}

File: kornell-core/src/main/java/kornell/core/shared/data/Enrollment.java
package kornell.core.shared.data;

import java.math.BigDecimal;
import java.util.Date;

public interface Enrollment {
	String getUUID();
	void setUUID(String uuid);
	Date getEnrolledOn();
	void setEnrolledOn(Date enrolledOn);
	String getCourseUUID();
	void setCourseUUID(String courseUUID);
	String getPersonUUID();
	void setPersonUUID(String personUUID);	
	/**
	 * @return Progress on the course as a percentage (between 0 and 1) 
	 */
	BigDecimal getProgress(); 
	void setProgress(BigDecimal progress);
	String getNotes();
	void setNotes(String notes);	
}
